compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -386,7 +386,40 @@ class OMR_EXTENSIBLE CodeGenerator
 
    TR_HasRandomGenerator randomizer;
 
-   bool supportsAtomicAdd() {return false;}
+   /** \brief
+    *     Determines whether intrinsics on the following atomic symbols is supported:
+    *
+    *     - atomicAdd32BitSymbol
+    *     - atomicAdd64BitSymbol
+    *     - atomicFetchAndAdd32BitSymbol
+    *     - atomicFetchAndAdd64BitSymbol
+    *
+    *  \return
+    *     \c true if intrinsics are supported; \c false otherwise.
+    */
+   bool supportsAtomicAdd()
+      {
+      return false;
+      }
+
+   /** \brief
+    *     Determines whether intrinsics on the following atomic symbols is supported:
+    *
+    *     - atomicSwap32BitSymbol
+    *     - atomicSwap64BitSymbol
+    *     - atomicCompareAndSwap32BitSymbol
+    *     - atomicCompareAndSwap64BitSymbol
+    *     - atomicCompareAndSwapReturnValue32BitSymbol
+    *     - atomicCompareAndSwapReturnValue64BitSymbol
+    *
+    *  \return
+    *     \c true if intrinsics are supported; \c false otherwise.
+    */
+   bool supportsAtomicSwap()
```

Comment:
Similar to above, hardware support for "atomic swap" and "atomic compare and swap" varies, so that codegen may not support both.
In addition, "atomic swap" and "atomic compare and swap" are conceptional different as the later one is a conditional operation while the other isn't.
If we believe a generic query for each symbol isn't necessary, I would still request to separate the queries for "atomic swap" and "atomic compare and swap".